shared/common/ui/select/index.tsx

```diff
@@ -179,6 +183,32 @@ export function Select<T extends any>({
     }
   }, [dropdownOpen]);
 
+  useKeyboardShortcut("p", () => setDropdownOpen((prev) => !prev));
+
+  function navigateFilter(e: React.KeyboardEvent<HTMLInputElement>) {
+    if (e.key === "Escape") {
+      return setDropdownOpen(false);
+    } else if (e.key === "Enter" && highlightedItem) {
+      setDropdownOpen(false);
+      return onChange?.(highlightedItem);
+    } else if (e.key === "ArrowUp" || e.key === "ArrowDown") {
+      const applicableList: SelectItem[] = filteredItems
+        ? filteredItems.map((item) => item.obj.item.item)
+        : (flattenedItems || [])
+            .filter((item) => item.type === "item")
+            .map((item) => item.item);
+
+      setHighlightedItem((prev) => {
+        const prevSelectedItemIndex = applicableList.findIndex(
+          (item) => item.id === prev?.id
+        );
+        return applicableList.at(
+          prevSelectedItemIndex + (e.key === "ArrowDown" ? 1 : -1)
```

Comment:
```suggestion
          (prevSelectedItemIndex + (e.key === "ArrowDown" ? 1 : -1)) % applicableList.length
```
Just so it wraps around at the end.